provisioner/windows-restart/provisioner.go

```diff
@@ -94,16 +96,42 @@ func (p *Provisioner) Provision(ui packer.Ui, comm packer.Communicator) error {
 		return fmt.Errorf("Restart script exited with non-zero exit status: %d", cmd.ExitStatus)
 	}
 
-	return waitForRestart(p)
+	return waitForRestart(p, comm)
 }
 
-var waitForRestart = func(p *Provisioner) error {
+var waitForRestart = func(p *Provisioner, comm packer.Communicator) error {
 	ui := p.ui
 	ui.Say("Waiting for machine to restart...")
 	waitDone := make(chan bool, 1)
 	timeout := time.After(p.config.RestartTimeout)
 	var err error
 
+	p.comm = comm
+	var cmd *packer.RemoteCmd
+	trycommand := TryCheckReboot
+	abortcommand := AbortReboot
+	// Stolen from Vagrant reboot checker
+	for {
+		log.Printf("Check if machine is rebooting...")
+		cmd = &packer.RemoteCmd{Command: trycommand}
+		err = cmd.StartWithUi(comm, ui)
+		if err != nil {
+			// Couldnt execute, we asume machine is rebooting already
+			break
+		}
+		if cmd.ExitStatus != 0 {
```

Comment:
IMO, this should check to see if `cmd.ExitStatus == 1115` since 1115 is the error code for "Already in progress". Other error codes should properly be shown as a build-terminating error.